src/usbhost/device-usbhost.cpp

```diff
@@ -0,0 +1,93 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#ifdef RS2_USE_ANDROID_BACKEND
+
+#include "device-usbhost.h"
+#include "endpoint-usbhost.h"
+#include "interface-usbhost.h"
+#include "usbhost.h"
+#include "../types.h"
+
+#include <string>
+#include <regex>
+#include <sstream>
+#include <mutex>
+
+namespace librealsense
+{
+    namespace platform
+    {
+        usb_device_usbhost::usb_device_usbhost(::usb_device* handle) :
+            _handle(handle)
+        {
+            usb_descriptor_iter it;
+            ::usb_descriptor_iter_init(handle, &it);
+            usb_descriptor_header *h = usb_descriptor_iter_next(&it);
+            librealsense::platform::usb_spec conn_spec;
+            if (h != nullptr && h->bDescriptorType == USB_DT_DEVICE) {
+                usb_device_descriptor *device_descriptor = (usb_device_descriptor *) h;
+                conn_spec = librealsense::platform::usb_spec(device_descriptor->bcdUSB);
+            }
+
+            do {
+                h = usb_descriptor_iter_next(&it);
+                if(h == NULL)
+                    break;
+                if (h->bDescriptorType == USB_DT_INTERFACE_ASSOCIATION) {
```

Comment:
Not clear - is it a leftover ?

Reply:
it's scanning all interfaces in a single USB configuration